### ISSUE TITLE
Automatically add a semicolon?

### ISSUE THREAD

**Issue by flashultra** [vshaxe/vshaxe#276]
Is it possible vshaxe automatically  to add a semicolon? 
In Eclipse have a option to inserts the semicolon at the end of the line when type it.
In Emacs can be used a macro (meta-Enter ) which, when pressed, would add a semicolon to the end of the current line and then place in the next line, at the proper indentation

**Comment by Gama11** [comment]
>In Eclipse have a option to inserts the semicolon at the end of the line when type it.

When you type what? Enter?

**Comment by flashultra** [comment]
No. At example I have function reset(boolean active) , so when I call the function I start typing:

**Comment by flashultra** [comment]
Here is example:
![Example](https://media.giphy.com/media/58ETjO6JjO3JUIo0sR/giphy.gif)

**Comment by Gama11** [comment]
Ah, I see. So it's less about the semicolon being inserted automatically, just that it's smart about figuring out where it should be placed.

**Comment by flashultra** [comment]
Yes, exactly.  Other way will be to use key combination ( <some key> + Enter ) to type semicolon at the end of line and to place cursor on new line, but I think Eclipse way ( for smart position of cursor)  is enough.

**Comment by flashultra** [comment]
Eclipse work in that way ( if option Semicolon is selected from Preferences)
 1) Type semicolon , if at the end of the row miss semicolon, move cursor at the end and add semicolon 
 2) If semicolon already exist at the end of the row and we type ";" in the middle of row, eclipse just print semicolon on the position on the cursor ( i.e. ignore 1) , because we already have ; at the end)
Maybe will be more useful if after 1) cursor position to be on new row ( i.e simulation on ; and Enter )
